docs/Debugging.md

```diff
@@ -0,0 +1,81 @@
+Debugging
+=========
+
+SGX-LKL provides a wrapper around gdb called `sgx-lkl-gdb`.
+This wrapper is compatible with IDEs and can be used instead of `gdb`.
+`sgx-lkl-gdb` automatically loads the SGX-LKL gdb plugin which
+ensures that debug symbols (if available) are loaded correctly. 
+When running in hardware mode, `sgx-lkl-gdb` uses the corresponding SGX debug
+instructions to read from and write to enclave memory.
+
+## Prerequisites
+
+### Open Enclave
+
+`sgx-lkl-gdb` relies on the Open Enclave gdb plugin.
+[Install Open Enclave](https://github.com/openenclave/openenclave/blob/master/docs/GettingStartedDocs/install_oe_sdk-Ubuntu_18.04.md) if necessary before continuing.
+
+The instructions below assume that Open Enclave was installed in the default directory `/opt/openenclave`.
+If you used a different directory, run `. <OE_PREFIX>/share/openenclave/openenclaverc` first.
+
+### Build mode
+
+Debugging enclave code (SGX-LKL as well as the application code) is only possible in debug and non-release builds of SGX-LKL.
+
+### Encrypted disks
+
+TODO add note on how to handle situations when disk keys are retrieved from a remote service that would not release them to debug/non-release builds of SGX-LKL (either re-create disks unencrypted, or, if available locally, hard-code key in app config for local debugging)
```

Comment:
We should be able to debug the key release process. Ideally, the key vault should have a "debugger friendly" release policy. During dev/debug cycles, the disk encryption keys should be created with such policies.

Reply:
Agree, good point. I changed the TODO.